enhancements/network/multi-network-vrfs.md

```diff
@@ -0,0 +1,924 @@
+used to mark egress IP packets and signify to br-ex which egress IP to SNAT to. An example where the egress IP is
+1.1.1.1 that maps to pkt_mark 10 would look something like this:
+
+![Egress IP VRF SGW](images/egress-ip-vrf-sgw.svg)
+
+For layer 2, egress IP has never been supported before. With the IC design, there is no need to have an
+ovn_cluster_router and join switch separating the layer 2 switch network (transit switch) from the GR. For non-IC cases
+this might be necessary, but for OpenShift purposes will only describe the behavior of IC in this proposal. In the layer
+2 IC model, GRs per node on a network will all be connected to the layer 2 transit switch:
+
+![Egress IP Layer 2](images/egress-ip-l2-primary.svg)
+
+In the above diagram, Node 2 is chosen to be the egress IP node for any pods in namespace A. Pod 1 and Pod 2 have
+default gateway routes to their respective GR on their node. When egress traffic leaves Pod 2, it is sent towards its
+GR-A on node 2, where it is SNAT’ed to the egress IP and the traffic sent to br-ex. For Pod 1, its traffic is sent to
+its GR-A on Node 1, where it is then rerouted towards GR-A on Node 2 for egress.
+
+##### Egress Firewall
+
+Egress firewall is enforced at the OVN logical switch, and this proposal has no effect on its functionality.
+
+##### Egress QoS
+
+Egress QoS is namespace scoped and functions by marking packets at the OVN logical switch, and this proposal has no
+effect on its functionality.
+
+##### Egress Service
+
+Egress service is namespace scoped and its primary function is to SNAT egress packets to a load balancer IP. As
+previously mentioned, the feature works the same in shared and local gateway mode, by leveraging the local gateway mode
+path. Therefore, its design will be covered in the Local Gateway Mode section of the Design Details.
+
+##### Multiple External Gateways (MEG)
+
+There will be no support for MEG or pod direct ingress on any network other than the primary, cluster default network.
+This support may be enhanced later by extending VRFs/networks outside the cluster.
+
+#### Local Gateway Mode
+
+With local gateway mode, egress/ingress traffic uses the kernel’s networking stack as a next hop. OVN-Kubernetes
+leverages an interface named “ovn-k8s-mp0” in order to facilitate sending traffic to and receiving traffic from the
+host. For egress traffic, the host routing table decides where to send the egress packet, and then the source IP is
+masqueraded to the node IP of the egress interface. For ingress traffic, the host routing table steers packets destined
+for pods via ovn-k8s-mp0 and SNAT’s the packet to the interface address.
+
+For multiple networks to use local gateway mode, some changes are necessary. The ovn-k8s-mp0 port is a logical port in
+the OVN topology tied to the cluster default network. There will need to be multiple ovn-k8s-mp0 ports created, one per
```

Comment:
the ovn-k8s-mp0 port is a veth that we create and attached to the node switch. We need 1 per network, because for local gw mode each switch needs a port to serve egress. That is for layer 2 and layer 3. In layer 2, the GR will reroute the packet out its mp0 for egress. You can think of ovn-worker-net2 as the transit switch for a layer 2 network. They are really the same thing, just a switch that spans multiple nodes. Either way for each network type, this per node part of the topology will look the same.

Reply:
AHHH gotcha, yeah now it makes more sense. thanks tim, I admit I haven't paid close attention to l2 stuff before this enhancement.